src/Adapter/Addons/AddonsDataProvider.php

```diff
@@ -151,6 +151,14 @@ public function request($action, $params = array())
             case 'check_module':
                 $post_data .= '&method=check&module_name='.urlencode($params['module_name']).'&module_key='.urlencode($params['module_key']);
                 break;
+            case 'module_download':
```

Comment:
Previously, the `module` case was used to download a module. With your new API, it allows us to get module details.

Instead of erasing your work, I brought back the module download in another case.